core/dns_test.go

```diff
@@ -14,8 +17,97 @@ import (
 	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/miekg/dns"
 )
 
+const dnsLoopbackAddr = "127.0.0.1:4053"
+
+func mockDNSQuery(w dns.ResponseWriter, r *dns.Msg) {
+	defer w.Close()
+	m := new(dns.Msg)
+	m.SetReply(r)
+	m.Compress = false
+
+	for _, q := range r.Question {
+		switch q.Qtype {
+		case dns.TypeSOA:
+			record := new(dns.SOA)
+			record.Hdr = dns.RR_Header{Name: "lets-encrypt.org.", Rrtype: dns.TypeSOA, Class: dns.ClassINET, Ttl: 0}
+			record.Ns = "ns.lets-encrypt.org."
+			record.Mbox = "master.lets-encrypt.org."
+			record.Serial = 1
+			record.Refresh = 1
+			record.Retry = 1
+			record.Expire = 1
+			record.Minttl = 1
+
+			m.Answer = append(m.Answer, record)
+			w.WriteMsg(m)
+			return
+		case dns.TypeA:
+			switch q.Name {
+			case "cps.letsencrypt.org.":
+				record := new(dns.A)
+				record.Hdr = dns.RR_Header{Name: "cps.lets-encrypt.org.", Rrtype: dns.TypeA, Class: dns.ClassINET, Ttl: 0}
```

Comment:
Was it intentional to use the `lets-encrypt.org` here in the RR_Header when the A query was for `letsencrypt.org` ?